cloudplatform/connectivity-destination-service/src/test/java/com/sap/cloud/sdk/cloudplatform/connectivity/DestinationServiceTest.java

```diff
@@ -1218,21 +1219,27 @@ void testConcurrentFetchSameDestinationButDifferentTenant()
         secondThread.join();
         firstThread.join();
 
-        final CacheKey firstTenantCacheKey = CacheKey.fromIds("TenantA", null).append(destinationName, options);
-        final CacheKey secondTenantCacheKey = CacheKey.fromIds("TenantB", null).append(destinationName, options);
+        // assert cache isolation locks are created for each tenant, for both get-single and get-all commands
+        assertThat(DestinationService.Cache.isolationLocks().asMap())
+            .containsOnlyKeys(
+                CacheKey.fromIds("TenantA", null).append(destinationName, options),
+                CacheKey.fromIds("TenantA", null).append(options),
+                CacheKey.fromIds("TenantB", null).append(destinationName, options),
+                CacheKey.fromIds("TenantB", null).append(options));
 
-        assertThat(DestinationService.Cache.isolationLocks()).isNotNull();
-        assertThat(DestinationService.Cache.isolationLocks().estimatedSize()).isEqualTo(2L);
-        assertThat(DestinationService.Cache.isolationLocks().getIfPresent(firstTenantCacheKey)).isNotNull();
-        assertThat(DestinationService.Cache.isolationLocks().getIfPresent(secondTenantCacheKey)).isNotNull();
+        // assert cache entries for one get-single command for each tenant
+        assertThat(DestinationService.Cache.instanceSingle().asMap())
+            .containsOnlyKeys(
+                CacheKey.fromIds("TenantA", null).append(destinationName, options),
+                CacheKey.fromIds("TenantB", null).append(destinationName, options));
 
-        assertThat(DestinationService.Cache.instanceSingle().estimatedSize()).isEqualTo(2L);
-        assertThat(DestinationService.Cache.instanceSingle().getIfPresent(firstTenantCacheKey)).isNotNull();
-        assertThat(DestinationService.Cache.instanceSingle().getIfPresent(secondTenantCacheKey)).isNotNull();
+        // assert no cache entries for get-all commands
+        assertThat(DestinationService.Cache.instanceAll().asMap()).isEmpty();
 
         verify(scpCfDestinationServiceAdapter, times(2))
             .getConfigurationAsJson("/destinations/" + destinationName, OnBehalfOf.TECHNICAL_USER_CURRENT_TENANT);
-
+        verify(scpCfDestinationServiceAdapter, never()) // get-all command not executed due to token exchange strategy
+            .getConfigurationAsJson(argThat(path -> !path.startsWith("/destinations/")), any());
```

Comment:
[minor]

What do you think about the following?
(Also applies to other tests that have been changed)

```suggestion
        // we are not performing further requests (i.e. there are no 'get-all' requests)
        verifyNoMoreInteractions(scpCfDestinationServiceAdapter);
```